### ISSUE TITLE
Research: Split diagram.js

### ISSUE THREAD

**Issue by b21phiro** [HGustavs/LenaSYS#13976]
## Description
We would like to split this file in to 3 parts, however we are unsure if we can. 

## Job
See if it's possible. If not, find out how we could make it possible. If we should split it in 3 parts, we would want to split the file atleast 50%.

**Comment by b21oscpe** [comment]
Is there any documentation of the code where we can get an overview of which parts can be split up? I suppose you have already researched a modular approach, are there any conclusions that has been made regarding it?

**Comment by b21eriho** [comment]
You are the research that you are assuming has already been made. You shouldn't change anything in the files at all, simply investigate if doing so would make sense.

**Comment by b21oscpe** [comment]
We have concluded that it is probably not impossible however it would require a lot of reworking of the code in order to split it up into several files. We have researched and tested a modular approach however it found no success as diagram.js functions has a lot of dependencies such as global variables and other functions. 

In order for this to work we need to find a solution to global variables and widely used functions.
